Add documentation for authentication token expirations (#321)
hashicorp/terraform-docs-common: website/docs/cloud-docs/users-teams-organizations/api-tokens.mdx

| Modify organizations               |      🔶     |             |                     |
| Manage organization tokens         |      🔶     |             |                     |
| View audit trails                  |             |             |          🔵         |

> | Invite users to organization       |      🔶     |      🔶     |          🔵         |

Reviewer: Great catch! 🕵️‍♂️

| Manage run tasks                   |      🔶     |      🔶     |          🔵          |
| **Modules**                        |             |             |                     |
| Manage Terraform modules           |      🔶     | 🔵 (owners) |                     |

## Token Expiration

> User, team, and organization tokens may be created with an expiration date and time set. Once this time has passed, the token will no longer be treated as valid, and may not be used to authenticate to any API.

Reviewer: Non-blocking suggestion: I think we can omit the "set" here. Additionally, what if we explicitly state that API requests will fail if made with an expired token (even though it's implied) -- thoughts?:
```suggestion
User, team, and organization tokens may be created with an expiration date and time. Once this time has passed, the token will no longer be treated as valid, and may not be used to authenticate to any API. Any API requests made with an expired token will result in failure.
```

> HashiCorp recommends setting an expiration on all new authentication tokens. Creating tokens which will expire once they are no longer needed helps reduce the risk that valid tokens may be leaked, or that token owners will forget to delete tokens meant for delegated use once their intended purpose is complete.

Reviewer: Non-blocking suggestion: What if we rephrase this slightly?:
```suggestion
HashiCorp recommends setting an expiration on all new authentication tokens. Creating tokens with an expiration date helps reduce the risk that valid tokens may be leaked, or that token owners will forget to delete tokens meant for delegated use once their intended purpose is complete.
```
